### PR TITLE
Finish snark test dev

### PR DESCRIPTION
**Pull request by QuiteStochastic**
develop branch version of https://github.com/MinaProtocol/mina/pull/13085

### CODE REVIEW

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Review by stevenplatt — APPROVED** [review]

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Review by deepthiskumar — APPROVED** [review]
Approving with the agreement of reducing # workers for payments_test.ml to 4. I think we should reduce in block_production_priority.ml as well, from 25 to 4 and reduce the scan state size (from 128 to 4)
Please keep the same for others (they are lower already and nonce test (with a different scan state size) is working well with 5 workers).

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Review by mrmr1993 — APPROVED** [review]
CODEOWNERS stamp, otherwise unreviewed

File: buildkite/scripts/run-test-executive.sh
Comment (deepthiskumar) [inline buildkite/scripts/run-test-executive.sh:8]: can we change this back to `opt-`? makes it easy to add more optional tests. Also, I thought these would run but not be required for PR merging and perhaps are candidates for nightly runs? I'm ok to create an issue and do it another PR.

Comment (QuiteStochastic) [reply buildkite/scripts/run-test-executive.sh:8]: cc: @georgeee. block prod prio has always been fully skipped in all PRs because it takes too many resources and takes too long.  I think the idea is that it should be run when needed, perhaps in nightly runs, but not every single PR (not even as an optional test net required for merging, because that would still take up resources to run it in the first place)

thanks for merging it anyways.

i changed the name because i felt like the "opt" prefix was a little hacky, and it was confusing because the name of the test in BK was inconsistent with the name of the test in the actual code.  

Honestly I'm not sure why we even have this `if` condition in `buildkite/scripts/run-test-executive.sh` instead of simply commenting out the `block-prod-prio` test in `buildkite/src/Jobs/Test/TestnetIntegrationTests.dhall`.

As far as nightlies, no lucy tests are currently run in the nightlies as far as i can see 

Screenshot from 2023-05-11 17-34-28

**Pull request merged**
